shotgun_api3/shotgun.py

```diff
@@ -1752,8 +1794,24 @@ def _response_errors(self, sg_response):
         """
 
         if isinstance(sg_response, dict) and sg_response.get("exception"):
-            raise Fault(sg_response.get("message",
-                "Unknown Error"))
+            
+            # handle the following cases explicitly
+            #
+            # {'exception': True, 
+            #  'error_code': 102, 
+            #  'message': "Can't authenticate session token 'xxxxxxxx'"}
+            #
+            # {'exception': True, 
+            #  'error_code': 102, 
+            #  'message': "Session token 'xxxxxxxx has expired'"}
+            
+            msg = sg_response.get("message")
+            
+            if msg and ("Can't authenticate session token" in msg or "has expired" in msg):
+                raise SessionAuthenticationFault(msg)
```

Comment:
Relying on the error message does feel a little iffy, it would be pretty easy for a change in the Shotgun codebase to break this and we wouldn't necessarily realize it.  At the very least there should be a comment in the Shotgun code that the python API depends on the wording of this errror.